webclient/src/components/admin/Category.js

```diff
@@ -1,30 +1,31 @@
 import React, { Component, useEffect, useState } from 'react';
-import useFetch from 'hooks/useFetch';
-import useMutation from 'hooks/useMutation';
 import M from 'materialize-css';
 import { Link } from 'react-router-dom';
+import { useQuery, useMutation, queryCache } from 'react-query';
+
 import FileUpload from './utils/FileUpload';
+import { client } from '~/utils/client';
+import { QUERY_KEY_ADMIN_CATEGORY } from '~/constants/query-keys';
+import { API_ADMIN_CATEGORY } from '~/constants/api-url';
 
 const Category = props => {
-  const [
-    { response: categories, loading: fetchLoading, error: fetchError },
-    doFetch
-  ] = useFetch({ endpoint: '/api/admin/categories' });
-  const [{ error: mutationError }, upsertData] = useMutation();
-
-  const handleRemove = async slug => {
-    upsertData({
-      method: 'delete',
-      endpoint: `/api/admin/categories/${slug}`
+  const {
+    data: categories,
+    status: fetchStatus,
+    error: fetchError
+  } = useQuery(QUERY_KEY_ADMIN_CATEGORY, () => client(API_ADMIN_CATEGORY));

```

Comment:
Hmm... I think by default the result of useQuery is cached for 5 minutes, perhaps this could cause an update to not be reflected directly?

Reply:
I think it is reflected directly, because it will refetch after a mutation, see line 19

Reply:
Oh I see, you use the queryCache there haha

Reply:
only after a successful mutation, otherwise it wont fetch..
but react-query by default auto refetch when you refocus to the window tho

Reply:
yeah but the refetch uses the cached result (with I think by default 5 minutes cache time iirc)